### PR TITLE
Run R CMD check on r-hub

### PR DESCRIPTION
**Pull request by gaborcsardi**
First shot at it.

### CODE REVIEW

File: R/r-hub.R [hunk 1]
```diff
@@ -0,0 +1,67 @@
+
+#' Build and check packag on r-hub
```
Comment (jimhester) [inline R/r-hub.R]: Missing an e in package.

File: R/r-hub.R [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                       interactive = FALSE, ...) {
+
+  pkg <- as.package(pkg)
+
+  built_path <- build(pkg, tempdir(), quiet = !interactive, ...)
+  on.exit(unlink(built_path), add = TRUE)
+
+  status <- rhub::check(
+    built_path,
+    email = maintainer(pkg)$email,
```
Comment (jimhester) [inline R/r-hub.R]: Maybe we should allow setting the email as an extra parameter? I know there have been a handful of times I wanted to check a package on win-builder I was not the normal maintainer for.

It also would be handy for checking PR before sending them.

File: DESCRIPTION [hunk 3]
```diff
@@ -26,7 +26,8 @@ Imports:
     jsonlite,
     stats,
     git2r (>= 0.11.0),
-    withr
+    withr,
+    rhub
```
Comment (jimhester) [inline DESCRIPTION]: Don't know how heavy-weight rhub is, but maybe this should be in `Suggests()` with a `devtools::check_suggested()`.

File: R/r-hub.R (hunk 1 above)
Comment (gaborcsardi) [inline R/r-hub.R]: Thanks! My son jumped on the 'e' key and it is half-broken now. :)

File: R/r-hub.R (hunk 2 above)
Comment (gaborcsardi) [inline R/r-hub.R]: I thought about that, and `rhub::check()` allows that. But `build_win()` does not, so I went with a similar API. But I can easily add this.

File: DESCRIPTION (hunk 3 above)
Comment (gaborcsardi) [inline DESCRIPTION]: Good point, I will do this, too. It is pretty lightweight, except for packages that are only needed by devtools: `httr` and `curl`. But if `testthat` is not required, then `rhub` should not be, either.

**Comment by gaborcsardi** [comment]
TODO list:

- [ ] add `email` argument
- [ ] make `rhub` package optional

**Comment by gaborcsardi** [comment]
Thanks for the suggestions, all should be fixed now.

**Comment by hadley** [comment]
Could you please add a bullet to NEWS?

File: DESCRIPTION [hunk 4]
```diff
@@ -26,7 +26,7 @@ Imports:
     jsonlite,
     stats,
     git2r (>= 0.11.0),
-    withr
+    withr,
```
Comment (hadley) [inline DESCRIPTION]: Trailing comma?

File: R/r-hub.R [hunk 5]
```diff
@@ -0,0 +1,73 @@
+
```
Comment (hadley) [inline R/r-hub.R]: Leading nl

**Comment by jimhester** [comment]
I merged this manually at https://github.com/hadley/devtools/commit/9a24447aad6ffe45c7143759ef510b0161c3c230, incoporating Hadley's comments.

**Pull request closed without merging**